pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -251,436 +168,179 @@ func TestInitFrameworkWithScorePlugins(t *testing.T) {
-						Score: 5,
-					},
-				},
+			name:    "ScoreWithNormalize plugin return score greater than MaxNodeScore",
+			plugins: buildConfig(scoreWithNormalizePlugin1),
+			injectedRes: map[string]injectedResult{
+				scoreWithNormalizePlugin1: {normalizeRes: MaxNodeScore + 1},
 			},
 			err: true,
 		},
 		{
-			name:    "Score plugin return score less than MinNodeScore",
-			plugins: plugin1And2,
-			input: PluginToNodeScores{
-				scorePlugin1: {
-					{
-						Name:  "node1",
-						Score: MaxNodeScore,
-					},
-					{
-						Name:  "node2",
-						Score: 3,
-					},
-				},
-				scorePlugin2: {
-					{
-						Name:  "node1",
-						Score: MinNodeScore - 1,
-					},
-					{
-						Name:  "node2",
-						Score: 5,
-					},
-				},
+			name:    "ScoreWithNormalize plugin return score less than MinNodeScore",
+			plugins: buildConfig(scoreWithNormalizePlugin1),
+			injectedRes: map[string]injectedResult{
+				scoreWithNormalizePlugin1: {normalizeRes: MinNodeScore - 1},
 			},
 			err: true,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			// Inject the results for each plugin.
+			registry := toRegistry(tt.injectedRes)
 			f, err := NewFramework(registry, tt.plugins, args)
 			if err != nil {
 				t.Fatalf("Failed to create framework for testing: %v", err)
 			}
 
-			status := f.ApplyScoreWeights(pc, pod, tt.input)
+			res, status := f.RunScorePlugins(pc, pod, nodes)
 
 			if tt.err {
 				if status.IsSuccess() {
-					t.Errorf("Expected status to be non-success.")
-				}
-			} else {
-				if !status.IsSuccess() {
-					t.Errorf("Expected status to be success.")
-				}
-				if !reflect.DeepEqual(tt.input, tt.want) {
-					t.Errorf("Score map after RunNormalizeScorePlugin: %+v, want: %+v.", tt.input, tt.want)
+					t.Error("Expected status to be non-success.")
 				}
+				return
+			}
+
+			if !status.IsSuccess() {
+				t.Errorf("Expected status to be success.")
+			}
+			if !reflect.DeepEqual(res, tt.want) {
+				t.Errorf("Score map after RunScorePlugin: %+v, want: %+v.", res, tt.want)
 			}
 		})
 	}
 }
+
+func toRegistry(injectedRes map[string]injectedResult) Registry {
+	registry := make(Registry)
+	for pl, f := range constructors {
+		npl := pl
+		nf := f
+		registry[pl] = func(_ *runtime.Unknown, _ FrameworkHandle) (Plugin, error) {
+			return nf(injectedRes[npl]), nil
+		}
+	}
+	return registry
+}
+
+func buildConfig(ps ...string) *config.Plugins {
+	var plugins []config.Plugin
+	for _, p := range ps {
+		plugins = append(plugins, config.Plugin{Name: p, Weight: weights[p]})
+	}
+	return &config.Plugins{Score: &config.PluginSet{Enabled: plugins}}
+}
+
+type injectedResult struct {
+	scoreRes     int
+	normalizeRes int
+	scoreErr     bool
+	normalizeErr bool
+}
+
+func injectScoreRes(inj injectedResult) (int, *Status) {
+	if inj.scoreErr {
+		return 0, NewStatus(Error, "injecting failure.")
+	}
+	return inj.scoreRes, nil
+}
+
+func injectNormalizeRes(inj injectedResult, scores NodeScoreList) *Status {
```

Comment:
`injectScoreRes` is used to get `scoreRes`, however `injectNormalizeRes ` is used to set scores. It is a little hard  to guess their functions from their name.

Reply:
Yeah I see your point. This is due to the semantics of the Score/NormalizeScore methods - Score returns a score value, while NormalizeScore mutates an existing score.

I renamed `injectScoreRes` to `setScoreRes` that may help. Let me know if you have other suggestions.